### PR TITLE
🎉 New Source: Merge [Low code CDK]

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Developing a new connector for the source: Merge*

Resolves:
[Merge](https://github.com/airbytehq/DeveloperWeek-Connector-Contest/issues/4)
Part of hackathon week

## How
*Developed using (Configuration Based Source) low-code CDK*

## Recommended reading order
1. `spec.yaml`
2. `manifest.yaml`
3. `schemas/*`

## Tests

<details open><summary><strong>Integration & Acceptance</strong></summary>
Full Test Results

image

</details>

## 🚨 User Impact 🚨
- No breaking changes, Just addition of new source
<details><summary><strong>New Connector: Source Merge</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

</details>

### CODE REVIEW

**Comment by btkcodedev** [comment]
PS for the reviewers: 
Source Merge is actually pulling data from other linked platforms. It is currently in the beta stages.
Thus most of the endpoints cannot be seeded with the sandbox account. (See screenshot of POST result)

image

Thus the implemented endpoints will give basic information about the merge and its integrated account.
Please ping via slack for sharing credentials

File: airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    primary_key: "model_id"
+    $parameters:
+      path: "/sync-status"
+
+  users_stream:
+    $ref: "#/definitions/base_stream"
+    name: "users"
+    primary_key: "id"
+    $parameters:
+      path: "/users"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml:64]: It is missing the incremental sync capability and pagination.

File: airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    authenticator:
+      type: BearerAuthenticator
+      api_token: "{{ config['api_token'] }}"
+
+  retriever:
+    type: SimpleRetriever
+    record_selector:
+      $ref: "#/definitions/selector"
+    paginator:
+      type: NoPagination
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml:25]: Some streams support pagination please check that.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
There are others endpoints. Are you planning to add them too? Left some comments of missing features.

File: airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml (hunk 1 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml:107]: Can you try to use a manual curl request? It doens't make sense to not work if the API docs says so.

Comment (btkcodedev) [reply airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml:107]: Tried with Postman, See remote_created_at field

image

**Comment by btkcodedev** [comment]
Seeding the endpoints through POST required paid plan, thus adding other streams as empty. 

image

File: airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml (hunk 1 above)
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml:107]: You need to have more records inside the users to have pagination and the documentations is saying to use created_after

Comment (btkcodedev) [reply airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml:107]: Uh. Little problem with that. See https://github.com/airbytehq/airbyte/pull/25342#issuecomment-1516473041

**Comment by btkcodedev** [comment]
Here are the credentials:
{
  "account_token": "kVpI78bNtlks9zhDXtaMtyYcpZquh07R2zdpRyHvurnc-JebX5fDcQ",
  "api_token": "ZqZvk2yty0ooFNKID8TBsEB13cFjtBFuXBbB0wwq6gV0Tr_Gihxy5g",
  "start_date": "2022-03-01T00:00:00.000Z"
}

No worries, Completely made up of temp email creds.

**Comment by btkcodedev** [comment]
The pagination feature added and other stream schemas are updated.
I couldn't seed the endpoints as it requires paid plan Tests are passing. 
Results below:

image

File: airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      $ref: "#/definitions/selective_paginator"
+    requester:
+      $ref: "#/definitions/requester"
+
+  base_stream:
+    type: DeclarativeStream
+    retriever:
+      $ref: "#/definitions/retriever"
+  
+  account_details_stream:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml:42]: Why you're not adding the `/accounts` endpoint?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml (hunk 3 above)
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-merge/source_merge/manifest.yaml:42]: The /accounts is for Accounting Base, the issue was inlt based for ATS URL
See https://docs.merge.dev/ats/overview/

All the endpoints inside Merge ATS is added :)

**Comment by btkcodedev** [comment]
@marcosmarxm Bump here :)

**Comment by btkcodedev** [comment]
@marcosmarxm Please add hackathon-developer-week label

**Review by marcosmarxm — APPROVED** [review]
Thanks @btkcodedev I'm having problems publishing the connector. Request help from our connector ops team. All your tasks are done!

**Comment by btkcodedev** [comment]
Thank you for your continued support & efforts @marcosmarxm

**Review by marcosmarxm — APPROVED** [review]
Thanks @btkcodedev

**Pull request merged**